bo/marktteilnehmer_test.go

```diff
@@ -0,0 +1,119 @@
+package bo
+
+import (
+	"encoding/json"
+	"github.com/corbym/gocrest/is"
+	"github.com/corbym/gocrest/then"
+	"github.com/go-playground/validator/v10"
+	"github.com/hochfrequenz/go-bo4e/com"
+	"github.com/hochfrequenz/go-bo4e/enum/anrede"
+	"github.com/hochfrequenz/go-bo4e/enum/botyp"
+	"github.com/hochfrequenz/go-bo4e/enum/geschaeftspartnerrolle"
+	"github.com/hochfrequenz/go-bo4e/enum/landescode"
+	"github.com/hochfrequenz/go-bo4e/enum/marktrolle"
+	"github.com/hochfrequenz/go-bo4e/enum/rollencodetyp"
+	"strings"
+)
+
+// Test_Marktteilnehmer_Deserialization deserializes an Marktteilnehmer json
+func (s *Suite) Test_Marktteilnehmer_Deserialization() {
+	var mt = Marktteilnehmer{
+		Marktrolle:       marktrolle.LF,
+		Makoadresse:      "edifact@my-favourite-marketpartner.de",
+		Rollencodetyp:    rollencodetyp.DVGW,
+		Rollencodenummer: "9903100000006",
+		Geschaeftspartner: Geschaeftspartner{
+			BusinessObject: BusinessObject{
+				BoTyp:             botyp.Geschaeftspartner,
+				VersionStruktur:   "1",
+				ExterneReferenzen: nil,
+			},
+			Anrede:               anrede.Divers,
+			Name1:                "Müller",
+			Name2:                "Lieschen",
+			Name3:                "",
+			Gewerbekennzeichnung: false,
+			HrNummer:             "handelsregister foo",
+			Amtsgericht:          "amtsgericht bar",
+			Kontaktweg:           0,
+			UmsatzsteuerId:       "umsatzsteuer foo",
+			GlaeubigerId:         "glauebiger bar",
```

Comment:
Gibt es eigentlich in diesen IDs auch Prüfziffern?
Zumindest die Gläubiger ID kennt eine Prüfziffer -> [wikipedia](https://de.wikipedia.org/wiki/Gl%C3%A4ubiger-Identifikationsnummer#Pr%C3%BCfziffer)

Okay die Umsatzsteuer-ID spielt in einer anderen Liga: [wikipedia](https://de.wikipedia.org/wiki/Umsatzsteuer-Identifikationsnummer#Aufbau_der_Identifikationsnummer)

Reply:
wäre dann eine separate FieldLevel Validation

Reply:
see #24 and #25